Source/DafnyCore/AST/IHasUsages.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+
+namespace Microsoft.Dafny;
+
+public interface IHasUsages : IDeclarationOrUsage {
+  public IEnumerable<IDeclarationOrUsage> GetResolvedDeclarations();
+}
+
+public interface ISymbol : IDeclarationOrUsage {
+  DafnySymbolKind Kind { get; }
+}
+
+public interface IHasSymbolChildren : ISymbol {
+  IEnumerable<ISymbol> ChildSymbols { get; }
+}
+
+public enum DafnySymbolKind {
```

Comment:
Is this enumeration based off some language server numbering? If please, please consider putting an URL to the specification so that we can keep it in sync and ignore my two questions below.